CriticalMass/RequestManager.swift

```diff
@@ -15,46 +15,69 @@ import os.log
 
 @available(macOS 10.12, *)
 public class RequestManager {
-    private struct SendLocationPostBody: Codable {
-        var device: String
-        var location: Location
-    }
-
     private struct SendMessagePostBody: Codable {
         var device: String
         var messages: [SendChatMessage]
     }
 
-    private var hasActiveRequest = false
-
     private var dataStore: DataStore
     private var locationProvider: LocationProvider
     private var networkLayer: NetworkLayer
     private var idProvider: IDProvider
+    private var networkObserver: NetworkObserver?
+
+    private let operationQueue: OperationQueue = {
+        let queue = OperationQueue()
+        queue.maxConcurrentOperationCount = 1
+
+        return queue
+    }()
 
     private var log = OSLog(subsystem: Bundle.main.bundleIdentifier ?? "", category: "RequestManager")
 
-    public init(dataStore: DataStore, locationProvider: LocationProvider, networkLayer: NetworkLayer, interval: TimeInterval = 12.0, idProvider: IDProvider) {
+    public init(dataStore: DataStore, locationProvider: LocationProvider, networkLayer: NetworkLayer, interval: TimeInterval = 12.0, idProvider: IDProvider, networkObserver: NetworkObserver?) {
         self.idProvider = idProvider
         self.dataStore = dataStore
         self.locationProvider = locationProvider
         self.networkLayer = networkLayer
-        configureTimer(with: interval)
-    }
+        self.networkObserver = networkObserver
 
-    private func configureTimer(with interval: TimeInterval) {
-        Timer.scheduledTimer(timeInterval: interval, target: self, selector: #selector(timerDidUpdate(timer:)), userInfo: nil, repeats: true)
+        setupNetworkObserver()
+        addUpdateOperation(with: interval)
     }
 
-    @objc private func timerDidUpdate(timer _: Timer) {
-        Logger.log(.info, log: log, "Timer did update")
-        updateData()
+    private func addUpdateOperation(with interval: TimeInterval) {
+        let updateDataOperation = UpdateDataOperation(locationProvider: locationProvider,
+                                                      idProvider: idProvider,
+                                                      networkLayer: networkLayer)
+        let taskIdentifier = UIApplication.shared.beginBackgroundTask {
```

Comment:
We might need to. add #if canImport(UIKit) for parts that uses UIKit Apis for  SPM compatibility

Reply:
fixed